test/eoftest/eoftest_runner.cpp

```diff
@@ -0,0 +1,78 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../utils/utils.hpp"
+#include "eoftest.hpp"
+#include <evmc/evmc.hpp>
+#include <evmone/eof.hpp>
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+namespace json = nlohmann;
+
+namespace
+{
+struct EOFValidationTest
+{
+    struct Case
+    {
+        struct Expectation
+        {
+            evmc_revision rev;
+            bool result;
+        };
+        std::string name;
+        evmc::bytes code;
+        std::vector<Expectation> expectations;
+    };
+    std::unordered_map<std::string, Case> cases;
+};
+
+void from_json(const json::json& j, EOFValidationTest::Case& o)
+{
+    const std::optional<evmc::bytes> hex_str{evmc::from_hex(j.at("code").get<std::string>())};
+    if (!hex_str)
+        throw std::invalid_argument{"invalid hex string"};
+    const evmc::bytes& code{hex_str.value()};
+    o.code = code;
+
+    for (const auto& [rev, result] : j.at("results").items())
+    {
+        o.expectations.emplace_back(to_rev(rev), result.at("result").get<bool>());
```

Comment:
Looks like not all compilers are happy with this, this should help I think
```suggestion
        o.expectations.emplace_back({to_rev(rev), result.at("result").get<bool>()});
```

Reply:
That doesn't seem to work either, so I decided to follow this code: https://github.com/ethereum/evmone/blob/095f83a3931c012af43e6580d188d4bb88af5d45/test/statetest/statetest_loader.cpp#L367

Looks like someone was faced with the same issue haha.